crm-project/src/main/java/CRM/utils/Validations.java

```diff
@@ -176,23 +185,37 @@ public static Long validateToken(String token) {
     }
 
 
-    public static <T> void setContentToFieldIfFieldExists(T object, String fieldName, Object content) throws NoSuchFieldException {
+    public static <T> void setContentToFieldIfFieldExists(T object, UpdateField fieldName, Object content) throws NoSuchFieldException {
+        String fieldNameModified = fieldName.toString().replaceAll("_","");
         try {
             for (Field field : object.getClass().getDeclaredFields()) {
                 field.setAccessible(true);
                 Object value = field.get(object);
-                if (value == null) {
-                    continue;
-                }
-                if (!field.getName().equals(fieldName)) {
+                if (!field.getName().equalsIgnoreCase(fieldNameModified)) {
                     continue;
                 }
-                if (!(value.getClass().equals(content.getClass()))) {
+                if (value != null && !(value.getClass().equals(content.getClass()))) {
                     value.getClass().cast(content);
                 }
                 field.set(object, content);
                 return;
             }
+
+            if(object.getClass().getSuperclass().equals(SharedContent.class)){
+                for (Field field : object.getClass().getSuperclass().getDeclaredFields()) {
+                    field.setAccessible(true);
+                    Object value = field.get(object);
+                    if (!field.getName().equalsIgnoreCase(fieldNameModified)) {
+                        continue;
+                    }
+                    if (value != null && !(value.getClass().equals(content.getClass()))) {
+                        value.getClass().cast(content);
+                    }
+                    field.set(object, content);
+                    return;
+                }
+            }
+
```

Comment:
Furthermore, both functions do exactly the same thing, except once it is called with parent entity and once with the actual entity. Make this a unique function that ones gets the actual entity class and if doesn't locate the field, call it again with parent.

For simplicity, you can make each of the functions above return a boolean, then you can know if the field was found or not.